### PR TITLE
planner/core: migrate test-infra to testify for 'planner/core/optimizer_test.go'

### LINKED ISSUE #28402: migrate test-infra to testify for `planner/core/optimizer_test.go`

### PR DESCRIPTION
**Pull request by johnhaxx7**
### What problem does this PR solve?

Issue Number: close #28402

Problem Summary:

### Check List

Tests 

- [x] Unit test

Side effects
NA

Documentation
NA

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/optimizer_test.go [hunk 1]
```diff
@@ -39,39 +36,40 @@ func testDecimalConvert(lDec, lLen, rDec, rLen int, lConvert, rConvert bool, cDe
 	rType.Flen = rLen
 
 	cType, lCon, rCon := negotiateCommonType(lType, rType)
-	c.Assert(cType.Tp, Equals, mysql.TypeNewDecimal)
-	c.Assert(cType.Decimal, Equals, cDec)
-	c.Assert(cType.Flen, Equals, cLen)
-	c.Assert(lConvert, Equals, lCon)
-	c.Assert(rConvert, Equals, rCon)
+	require.Equal(t, cType.Tp, mysql.TypeNewDecimal)
```
Comment (hawkingrei) [inline planner/core/optimizer_test.go:39]: ```require.Equal(t TestingT, expected interface{}, actual interface{})```

Please check the order of parameters.

**Review by hawkingrei — APPROVED** [review]

File: planner/core/optimizer_test.go (hunk 1 above)
Comment (johnhaxx7) [reply planner/core/optimizer_test.go]: Nice catch! Updated, thanks!

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**